shared_model/backend/protobuf/transaction_responses/proto_tx_response.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_PROTO_TX_RESPONSE_HPP
+#define IROHA_PROTO_TX_RESPONSE_HPP
+
+#include "backend/protobuf/transaction_responses/proto_concrete_tx_response.hpp"
+#include "utils/lazy_initializer.hpp"
+#include "utils/reference_holder.hpp"
+#include "utils/variant_deserializer.hpp"
+
+template <typename... T>
+auto load(const iroha::protocol::ToriiResponse &ar) {
+  unsigned which = ar.tx_status();
+  constexpr unsigned last = boost::mpl::size<T...>::type::value - 1;
+
+  return shared_model::detail::variant_impl<T...>::template load<
+      shared_model::interface::TransactionResponse::ResponseVariantType>(
+      ar, which > last ? last : which);
+  ;
+}
+
+namespace shared_model {
+  namespace proto {
+    /**
+     * TransactionResponse is a status of transaction in system
+     */
+    class TransactionResponse final : public interface::TransactionResponse {
+     private:
+      /// PolymorphicWrapper shortcut type
+      template <typename... Value>
+      using wrap = boost::variant<detail::PolymorphicWrapper<Value>...>;
```

Comment:
Maybe move it into common space?

Reply:
Maybe in other pr? Otherwise thing messed up

Reply:
Ok, we should keep it in mind.